### ISSUE TITLE
uv compile now shows annotations for -c and -r paths that causes issues cross platform

### ISSUE THREAD

**Issue by inoa-jboliveira** [astral-sh/uv#3800]
Since uv 0.2.x the pip compile tool adds -r and -c annotations on the "via" part of the code, but uses windows-style paths when running on Windows, not respecting the command line on "-c". 

tested both
uv 0.2.1 (1fc71ea73 2024-05-22)
uv 0.2.2 (e52ae0e2b 2024-05-22)

See diff below
![Screenshot 2024-05-23 154932](https://github.com/astral-sh/uv/assets/148236158/d3ac3456-41b8-45c5-bb0d-d398307a4891)

my command line is (I run from the project root):
`uv pip compile ./requirements/requirements.in -o ./requirements/requirements-win.txt --python-platform=windows`

I run the same command line on Windows and on my Linux CI, I make sure to always use forward slashes so outputs are compatible.

In my "requirements.in" file there is a line `-c dependencies.txt` since the file is in the same directory, I cannot prefix it with "-c requirements/dependencies.txt" so UV gets to chose how to build the path. 

I like having the annotations and wouldn't want to switch to "--no-annotate", but running the same command in 2 different OSs now accuses a diff in the files.

Since UV already respects the forward slashes when I pass "requirements/requirements.in", could it use the same prefix of the file containing the value when prefixing names that are relative?

**Comment by charliermarsh** [comment]
Hmm, I think the annotation should match whatever form is used verbatim. What is the exact form used in `requirements.in`?

**Comment by charliermarsh** [comment]
Can you include a `requirements.in` file that replicates it along with the command? Thanks!

**Comment by charliermarsh** [comment]
Oh sorry I see. You're using a relative path. Hmm...

**Comment by inoa-jboliveira** [comment]
Simplified version of my setup:

- /project-root/
   - requirements/
      - requirements.in
      - dependencies.txt
      - requirements-win.txt
  - compile.bat


requirements.in: 
```
-c dependencies.txt
bcrypt
bs4
```
dependencies.txt
```
pip>24
```

compile.bat (it contains just the commands so I use it both as a bat and sh file)
```
uv pip compile ./requirements/requirements.in -o ./requirements/requirements-win.txt --python-platform=windows
```

**Comment by inoa-jboliveira** [comment]
I would be ok with the -c being verbatim. Althought it woud be less correct i guess

**Comment by charliermarsh** [comment]
I agree that it would be nice to be consistent here. Will look into it.

**Comment by charliermarsh** [comment]
Marking as a bug.

**Comment by inoa-jboliveira** [comment]
I think this could be solved in the user_display method (or maybe the origin.path()). I am having issues with the rust debugger on Windows right now, but I think at this point it should be possible to acquire the path separator.

https://github.com/astral-sh/uv/blob/3e4365301e36ab084650a6acdbe2b7b1487777c2/crates/distribution-types/src/annotation.rs#L31-L36

https://github.com/astral-sh/uv/blob/bc963d13cb5b1e0ca0783f62848d3ff5d93f3dda/crates/uv-fs/src/path.rs#L52-L63

https://github.com/astral-sh/uv/blob/3e4365301e36ab084650a6acdbe2b7b1487777c2/crates/pep508-rs/src/origin.rs#L16-L21


Here is the original PR published in v0.1.43
- https://github.com/astral-sh/uv/pull/3269
